### PR TITLE
Refactors classes to sub-namespaces

### PR DESCRIPTION
**Pull request by dborsatto**
Classes and interfaces have been moved to sub-namespaces for improved organization, instead of having everything in the main `Contentful\Managament` namespace, as the number of classes keeps growing and things are becoming a bit unwieldy.

Entities managed by Contentful (_Asset_, _ContentType_, _Entry_, _EntrySnapshot_, _Locale_, _PublishedContentType_, _Space_, _Upload_, _User_, _Webhook_, _WebhookCall_, _WebhookCallDetails_, _WebhookHealth_) are now in the namespace `Contentful\Management\Resource`.

Behavioral interfaces (_Archivable_, _Creatable_, _Deletable_, _Publishable_, _Updatable_) are now in the namespace `Contentful\Management\Behavior`.

Changes have been made using PHPStorm's "move" functionality, and tests confirm that nothing's broken, so we're 99.9999% sure it's safe.

~As the 3 other PRs currently open (https://github.com/contentful/contentful-management.php/pull/27, https://github.com/contentful/contentful-management.php/pull/26, https://github.com/contentful/contentful-management.php/pull/7)  would be impacted by this, I say we could either merge this and rebase those (which would probably be the cleanest solution) or wait for those to be merged and then rebase this one against master.~

---
> Update: As #27 and #7 have been merged, I've rebased against master and updated this PR.

> Update 2: As discussed internally, `Resource` was preferred over `Entity`.

### CODE REVIEW

**Comment by dborsatto** [comment]
Update: As #27 and #7 have been merged, I've rebased against master and updated PR,

**Comment by mariobodemann** [comment]
Just a quick questions: `Resource` vs `Entity` why would you choose either?

**Comment by dborsatto** [comment]
@mariobodemann they both could be a valid alternative, especially in this context, but the term `Entity` is commonly used in one of the most used frameworks (Symfony), so I chose a term that already has solid roots in the PHP ecosystem.
`Resource` is also word that right now is safe, but I think it's one of those "at risk", which could become a reserved keyword somewhere along the way, whereas `Entity` because of its current status, should be safer...

**Comment by dborsatto** [comment]
Updated the PR using `Resource` instead of `Entity`.

File: src/Management/SpaceManager.php [hunk 1]
```diff
@@ -51,7 +68,7 @@ public function __construct(Client $client, ResourceBuilder $builder, $spaceId)
     }
 
     /**
-     * Checks that the given resource is compatible with the currently-managend space.
+     * Checks that the given resource is compatible with the currently-managed space.
```
Comment (mariobodemann) [inline src/Management/SpaceManager.php]: 😍

**Pull request merged**